Make Ansible compatible with new resolvelib

Based on:

ansible-galaxy - support resolvelib >= 0.5.3, < 1.1.0 #80196 
https://github.com/ansible/ansible/pull/80196/commits

diff -ruNp ansible-stable-2.14/changelogs/fragments/80196-resolvelib_lt_1_1_0.yml ansible-stable-2.14.new/changelogs/fragments/80196-resolvelib_lt_1_1_0.yml
--- ansible-stable-2.14/changelogs/fragments/80196-resolvelib_lt_1_1_0.yml	1970-01-01 01:00:00.000000000 +0100
+++ ansible-stable-2.14.new/changelogs/fragments/80196-resolvelib_lt_1_1_0.yml	2023-04-17 16:11:49.256649104 +0200
@@ -0,0 +1,2 @@
+minor_changes:
+  - ansible-galaxy - support ``resolvelib >= 0.5.3, < 1.1.0``.
diff -ruNp ansible-stable-2.14/lib/ansible/galaxy/dependency_resolution/providers.py ansible-stable-2.14.new/lib/ansible/galaxy/dependency_resolution/providers.py
--- ansible-stable-2.14/lib/ansible/galaxy/dependency_resolution/providers.py	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/lib/ansible/galaxy/dependency_resolution/providers.py	2023-04-17 16:14:05.090736055 +0200
@@ -42,7 +42,7 @@ except ImportError:
 
 # TODO: add python requirements to ansible-test's ansible-core distribution info and remove the hardcoded lowerbound/upperbound fallback
 RESOLVELIB_LOWERBOUND = SemanticVersion("0.5.3")
-RESOLVELIB_UPPERBOUND = SemanticVersion("0.9.0")
+RESOLVELIB_UPPERBOUND = SemanticVersion("1.1.0")
 RESOLVELIB_VERSION = SemanticVersion.from_loose_version(LooseVersion(resolvelib_version))
 
 
diff -ruNp ansible-stable-2.14/requirements.txt ansible-stable-2.14.new/requirements.txt
--- ansible-stable-2.14/requirements.txt	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/requirements.txt	2023-04-17 16:15:01.105054994 +0200
@@ -12,4 +12,4 @@ packaging
 # NOTE: Ref: https://github.com/sarugaku/resolvelib/issues/69
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.9.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 1.1.0  # dependency resolver used by ansible-galaxy
diff -ruNp ansible-stable-2.14/test/integration/targets/ansible-galaxy-collection/vars/main.yml ansible-stable-2.14.new/test/integration/targets/ansible-galaxy-collection/vars/main.yml
--- ansible-stable-2.14/test/integration/targets/ansible-galaxy-collection/vars/main.yml	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/test/integration/targets/ansible-galaxy-collection/vars/main.yml	2023-04-17 16:11:49.258727304 +0200
@@ -9,6 +9,8 @@ supported_resolvelib_versions:
   - "0.6.0"
   - "0.7.0"
   - "0.8.0"
+  - "0.9.0"
+  - "1.0.1"
 
 unsupported_resolvelib_versions:
   - "0.2.0"  # Fails on import
diff -ruNp ansible-stable-2.14/test/lib/ansible_test/_data/requirements/ansible.txt ansible-stable-2.14.new/test/lib/ansible_test/_data/requirements/ansible.txt
--- ansible-stable-2.14/test/lib/ansible_test/_data/requirements/ansible.txt	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/test/lib/ansible_test/_data/requirements/ansible.txt	2023-04-17 16:16:50.079393176 +0200
@@ -12,4 +12,4 @@ packaging
 # NOTE: Ref: https://github.com/sarugaku/resolvelib/issues/69
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.9.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 1.1.0  # dependency resolver used by ansible-galaxy
diff -ruNp ansible-stable-2.14/test/sanity/code-smell/docs-build.requirements.in ansible-stable-2.14.new/test/sanity/code-smell/docs-build.requirements.in
--- ansible-stable-2.14/test/sanity/code-smell/docs-build.requirements.in	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/test/sanity/code-smell/docs-build.requirements.in	2023-04-17 16:15:20.096695677 +0200
@@ -1,6 +1,6 @@
 jinja2
 pyyaml
-resolvelib < 0.9.0
+resolvelib < 1.1.0
 sphinx == 4.2.0
 sphinx-notfound-page
 sphinx-ansible-theme
diff -ruNp ansible-stable-2.14/test/sanity/code-smell/docs-build.requirements.txt ansible-stable-2.14.new/test/sanity/code-smell/docs-build.requirements.txt
--- ansible-stable-2.14/test/sanity/code-smell/docs-build.requirements.txt	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/test/sanity/code-smell/docs-build.requirements.txt	2023-04-17 16:15:58.181616446 +0200
@@ -27,7 +27,7 @@ pyparsing==3.0.9
 pytz==2022.2.1
 PyYAML==6.0
 requests==2.28.1
-resolvelib==0.8.1
+resolvelib==1.0.1
 rstcheck==3.5.0
 semantic-version==2.10.0
 sh==1.14.3
diff -ruNp ansible-stable-2.14/test/sanity/code-smell/package-data.requirements.in ansible-stable-2.14.new/test/sanity/code-smell/package-data.requirements.in
--- ansible-stable-2.14/test/sanity/code-smell/package-data.requirements.in	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/test/sanity/code-smell/package-data.requirements.in	2023-04-17 16:16:26.423015900 +0200
@@ -1,7 +1,7 @@
 docutils < 0.18  # match version required by sphinx in the docs-build sanity test
 jinja2
 pyyaml  # ansible-core requirement
-resolvelib < 0.9.0
+resolvelib < 1.1.0
 rstcheck < 4  # match version used in other sanity tests
 straight.plugin
 antsibull-changelog
diff -ruNp ansible-stable-2.14/test/sanity/code-smell/package-data.requirements.txt ansible-stable-2.14.new/test/sanity/code-smell/package-data.requirements.txt
--- ansible-stable-2.14/test/sanity/code-smell/package-data.requirements.txt	2023-04-12 22:09:39.000000000 +0200
+++ ansible-stable-2.14.new/test/sanity/code-smell/package-data.requirements.txt	2023-04-17 16:15:42.681781780 +0200
@@ -6,7 +6,7 @@ MarkupSafe==2.1.1
 packaging==21.3
 pyparsing==3.0.9
 PyYAML==6.0
-resolvelib==0.8.1
+resolvelib==1.0.1
 rstcheck==3.5.0
 semantic-version==2.10.0
 straight.plugin==1.5.0
